src/common/slash.ts

```diff
@@ -0,0 +1,30 @@
+import { ApplicationCommandOptionData, CommandInteraction } from 'discord.js';
+import { IContainer } from './types';
+import { z } from 'zod';
+
+export const slashCommands: Map<string, ISlashCommand> = new Map();
+
+export const SlashCommand = z.object({
+  commandName: z.string(),
+  name: z.string(),
+  description: z.string(),
+  options: z.optional(z.array(z.any())),
+  execute: z
+    .function()
+    .args(z.any())
+    .returns(z.union([z.void(), z.promise(z.void())])),
+});
+
+export interface ISlashCommand {
```

Comment:
is there a reason why we couldn't just infer the typing from the zod object?

Reply:
The Zod typings for the `execute` method and `options` use Discord.js types that aren't in Zod and would be a pain to move over, and aren't fully descriptive. I could change to an inference + an intersection type, but I'm not sure that's better.

Reply:
Nah don't worry about moving over. that sounds like too much effort lmao